### DISCUSSION TITLE
Pixi and Emacs

### DISCUSSION THREAD

**Discussion by kklein** [prefix-dev/pixi#1830] [Q&A]
Hi!

I enjoy working with pixi - thanks a lot for the great work!

One shortcoming I currently stumble upon is my development workflow in Emacs. have been using [`pyvenv.el`](https://github.com/jorgenschaefer/pyvenv) when working with conda environments. Thanks to `pyvenv.el`, I can choose and activate a specific conda environment for my Emacs session. This brings two advantages:

- Dependencies can be found in light of static code analysis (including code completion) 
- I can use an (IPython) REPL to execute bits of code I select in my editor view

Now, working with pixi on some projects, I haven't yet found a way to get aforementioned benefits when working with Emacs. Hence my questions:

- Has anyone found a way to get these things to work with Emacs?
- If not, are there some plans to build an Emacs plugin as has been the case for e.g. [Visual Studio Code](https://github.com/prefix-dev/pixi/issues/411)?

**Comment by tdejager** [comment]
No plans for an emacs plugin yet. I think as a short term workaround. If you use pyright:

https://github.com/emacs-lsp/lsp-pyright

You can use the pyright venv settings, explained on zed docs (https://zed.dev/docs/languages/python) and set that directly, and you should get completions. Now if those could be set via lsp commands (i'm unsure), you might be able to use some elisp to switch to an environment if you want.
